### PR TITLE
Feature/refactor code (PR)

### PR DESCRIPTION
**Pull request by teke85**
# Project Requirements

- Made sure YAGNI, KISS, and DRY rules and Ruby style guide were respected.
- Followed SOLID principles.
- Payed special attention to the "Single-responsibility Principle".
- Made sure that the role of main.rb was only to start our code and ask the user for the initial choice of options (or to quit).
- Extracted some pieces of logic into separate class(es).

Team
@teke85 
@EddyShimwa

### CODE REVIEW

File: lib/app.rb [hunk 1]
```diff
@@ -74,6 +66,18 @@ def create_teacher
     puts 'Teacher Created Successfully'
   end
 
+  def create_person
```
Comment (gemmen29) [inline lib/app.rb:60]: - [x] Kindly move the logic for request user input in the following method: `create_person`, `create_student`, `create_teacher`, `create_book`, and `create_rental` to the main file or classes/methods whose responsibilities are to request or validate input as this violate the `Single responsibility rule` and `YAGNI and KISS`. You can read this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) as a reminder about `Best practices and Solid lesson`.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/app.rb (hunk 1 above)
Comment (EddyShimwa) [reply lib/app.rb:60]: Thanks @gemmen29 for reviewing this Pr

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EddyShimwa** [comment]
Thanks @gemmen29

**Pull request merged**